### PR TITLE
User can access it's profile page

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT-link](https://www.pivotaltracker.com/story/show/170646663)

- Adds UsersController
- Adds route
- Adds serializer
- Restricts show for profile page

### CODE REVIEW

File: app/controllers/api/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Api::Admin::UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    authorize(current_user)
+    user = User.find(params[:id])
```
Comment (tochman) [inline app/controllers/api/admin/users_controller.rb:5]: Why do you authorize for `current_user`?

File: config/routes.rb [hunk 2]
```diff
@@ -1,11 +1,18 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :admin do
+      get 'users/show'
+    end
+  end
+  get 'users/show'
```
Comment (tochman) [inline config/routes.rb]: This route needs to go. But I'm a bit confused, why do you have 2 namespaces with the same name?

File: spec/factories/users.rb [hunk 3]
```diff
@@ -12,5 +12,9 @@
       email { "user#{rand(1...9999)}@mail.com" }
       role { "publisher" }
     end
+    factory :subscriber do
+      email { "user#{rand(1...9999)}@mail.com" }
```
Comment (tochman) [inline spec/factories/users.rb]: If it's the same way to create an email, then it should be enough if you have it in the main factory, right?

File: spec/requests/api/admin/user/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    let!(:non_authorized_headers) { { HTTP_ACCEPT: 'application/json' } }
+    before do
+      get "/api/admin/users/#{user.id}", headers: non_authorized_headers
+    end
+
+    it 'returns a 401 response status' do
+      expect(response).to have_http_status 401
+    end
+  end
+end
```
Comment (tochman) [inline spec/requests/api/admin/user/show_spec.rb]: Please mind code formatting on this one.

File: app/controllers/api/admin/users_controller.rb (hunk 1 above)
Comment (mantanquzzmo) [reply app/controllers/api/admin/users_controller.rb:5]: We had a silly idea that it was needed for our restriction. Turned out that and the policies were redundant. Fixed

File: config/routes.rb (hunk 2 above)
Comment (mantanquzzmo) [reply config/routes.rb]: Unnoticed addition generated by rails. Removed.

File: spec/requests/api/admin/user/show_spec.rb (hunk 4 above)
Comment (mantanquzzmo) [reply spec/requests/api/admin/user/show_spec.rb]: Moved headers to top.

File: spec/factories/users.rb (hunk 3 above)
Comment (mantanquzzmo) [reply spec/factories/users.rb]: Fixed.

File: config/routes.rb (hunk 2 above)
Comment (tochman) [reply config/routes.rb]: I still see the extra route

**Review by tochman — APPROVED** [review]
Fantastic. Small, clean, right on target. 👊

**Pull request merged**